### PR TITLE
nimmi89/c04-iac01

### PR DESCRIPTION
**Pull request by nimmi89**
# Description

Exercise : C04-IAC01

Student Name:Nimmi Varghese

> Create EC2, S3 and access S3 bucket from EC2 using Terraform.

## Type of change

- [ ] Exercise Submission (one exercise per PR)
  - [ ] This PR is just for **one** exercise of a class
  - [ ] PR name follows the pattern `<github-username>/<exercise-number>`
  - [ ] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/04class/exercises/c04-iac01/nimmi89/ec2.tf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  tags = {
+     Name = "DA-Instance-Terraform"
+ }
+ 
+ user_data = <<-EOF
+              #!/bin/bash
+              yum update -y
+              yum install -y httpd
+              curl 169.254.169.254/latest/meta-data/hostname > index.html
+              EOF
```
Comment (alanlima) [inline classes/04class/exercises/c04-iac01/nimmi89/ec2.tf:17]: Any particular reason to add this script here?

File: classes/04class/exercises/c04-iac01/nimmi89/security-group.tf [hunk 2]
```diff
@@ -0,0 +1,14 @@
+resource "aws_security_group" "DA-sg"{
+    ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
```
Comment (alanlima) [inline classes/04class/exercises/c04-iac01/nimmi89/security-group.tf:6]: For security, use your home ip address here.

Comment (nimmi89) [reply classes/04class/exercises/c04-iac01/nimmi89/security-group.tf]: Yes. Missed that.I have updated the same

File: classes/04class/exercises/c04-iac01/nimmi89/ec2.tf (hunk 1 above)
Comment (nimmi89) [reply classes/04class/exercises/c04-iac01/nimmi89/ec2.tf]: Oh! I was trying how to pass user data through terraform for my reference. Forgot to take it off! I Have removed those lines!Thanks!

Comment (alanlima) [reply classes/04class/exercises/c04-iac01/nimmi89/ec2.tf]: No worries, nice that you are playing around with the options 👏👏

Now you can redo all the previous exercises using terraform 🤣🤣🤣

**Pull request merged**